src/marks/waffle.js

```diff
@@ -24,20 +26,20 @@ export class WaffleX extends BarX {
 }
 
 export class WaffleY extends BarY {
-  constructor(data, {unit = 1, gap = 1, round, render, multiple, ...options} = {}) {
-    super(data, {...options, render: composeRender(render, waffleRender("y"))}, waffleDefaults);
+  constructor(data, {unit = 1, gap = 1, round, multiple, ...options} = {}) {
+    super(data, wafflePolygon("y", options), waffleDefaults);
     this.unit = Math.max(0, unit);
     this.gap = +gap;
     this.round = maybeRound(round);
     this.multiple = maybeMultiple(multiple);
   }
 }
 
-function waffleRender(y) {
-  return function (index, scales, values, dimensions, context) {
-    const {ariaLabel, href, title, ...visualValues} = values;
-    const {unit, gap, rx, ry, round} = this;
-    const {document} = context;
+function wafflePolygon(y, options) {
+  return initializer(waffleRender(options), function (data, facets, channels, scales, dimensions) {
+    const {round, unit} = this;
+
+    const values = valueObject(channels, scales);
```

Comment:
This is passing all the channels through the scales, which is probably more than we need. I think we only need to materialize _e.g._ _x_, _y1_, _y2_.